lib/irb/pager.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module IRB
+  # The implementation of this class is borrowed from RDoc's lib/rdoc/ri/driver.rb.
+  # Please do NOT use this class directly outside of IRB.
+  class Pager
+    PAGE_COMMANDS = [ENV['RI_PAGER'], ENV['PAGER'], 'less', 'more'].compact.uniq
+
+    class << self
+      def page
+        if STDIN.tty? && pager = setup_pager
```

Comment:
I'm not sure if we should check `STDIN.tty?` or `STDOUT.tty?`. In most cases they should return the same value?

Reply:
:+1: It's consistent with IRB shows message `Switch to inspect mode.` when stdin is not a tty.


`irb` → both tty
`irb > output_file` → stdin is a tty
`cat input_file | irb` → stdout is a tty
`cat input_file | irb > output_file` → neither is tty